datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -228,7 +228,7 @@ impl OptimizerRule for PushDownProjection {
                 // Gather all columns needed for expressions in this Aggregate
                 let mut new_aggr_expr = vec![];
                 for e in agg.aggr_expr.iter() {
-                    let column = Column::from(e.display_name()?);
+                    let column = Column::from_name(e.display_name()?);
```

Comment:
This change was done in https://github.com/apache/arrow-datafusion/pull/7981 -- it doesn't seem necessary and I think was just a mistake (and did not have adequate test coverage)

Reply:
Thanks for resolving this; this was a leftover from my initial attempt to address the qualified alias problem (without altering the `Expr::Alias` enum) that I forgot to rollback.

Reply:
And sadly we had no test coverage for it :(